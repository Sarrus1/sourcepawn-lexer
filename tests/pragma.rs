mod utils;

use insta::assert_json_snapshot;
use sourcepawn_lexer::*;
use utils::collect_tokens;

#[test]
fn pragma_simple() {
    let input = r#"#pragma deprecated foo
"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_no_line_break() {
    let input = "#pragma deprecated foo";

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_trailing_line_comment() {
    let input = r#"#pragma deprecated foo //bar
"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_trailing_block_comment() {
    let input = r#"#pragma deprecated foo /* */
"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_with_block_comment() {
    let input = r#"#pragma deprecated foo /* */ bar
"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_with_block_comment_and_line_continuation() {
    let input = r#"#pragma deprecated foo /* */ \
bar
"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_with_trailing_multiline_block_comment() {
    let input = r#"#pragma deprecated foo /*
*/ bar
"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_with_trailing_line_continuated_multiline_block_comment() {
    let input = r#"#pragma deprecated foo /* \
*/ bar
"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_line_continuation() {
    let input = r#"#pragma deprecated foo \
bar
"#;

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_line_continuation_carriage_return() {
    let input = "#pragma deprecated foo \\\r\nbar\n";

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}

#[test]
fn pragma_unicode() {
    let input = "#pragma deprecated \"Устаревшая функция. Плагин автоматически очищает всё, что создал другой выгруженный плагин.\"";

    let mut lexer = SourcepawnLexer::new(input);
    assert_json_snapshot!(collect_tokens(&mut lexer));
}
