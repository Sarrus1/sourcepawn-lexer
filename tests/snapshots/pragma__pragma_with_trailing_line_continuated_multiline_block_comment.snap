---
source: tests/pragma.rs
assertion_line: 78
expression: collect_tokens(&mut lexer)
snapshot_kind: text
---
[
  {
    "kind": {
      "PreprocDir": "MPragma"
    },
    "text": "#pragma deprecated foo /* \\\n*/ bar",
    "range_start": 0,
    "range_end": 34,
    "delta": 0,
    "in_preprocessor": true
  },
  {
    "kind": "Newline",
    "text": "\n",
    "range_start": 34,
    "range_end": 35,
    "delta": 0,
    "in_preprocessor": false
  },
  {
    "kind": "Eof",
    "text": "\u0000",
    "range_start": 35,
    "range_end": 35,
    "delta": 0,
    "in_preprocessor": false
  }
]
